modules/contact/src/AugmentedLagrangianContactProblem.C

```diff
@@ -0,0 +1,167 @@
+                                                          snorm,
+                                                          fnorm,
+                                                          rtol,
+                                                          stol,
+                                                          abstol,
+                                                          nfuncs,
+                                                          my_max_funcs,
+                                                          ref_resid,
+                                                          my_div_threshold);
+
+  _console << "Augmented Lagrangian contact iteration " << _num_lagmul_iterations << "\n";
+
+  bool _augLM_repeat_step;
+
+  if (reason > 0)
+  {
+    if (_num_lagmul_iterations < _max_lagmul_iters)
+    {
+      NonlinearSystemBase & nonlinear_sys = getNonlinearSystemBase();
+      nonlinear_sys.update();
+
+      const ConstraintWarehouse & constraints = nonlinear_sys.getConstraintWarehouse();
+
+      std::map<std::pair<unsigned int, unsigned int>, PenetrationLocator *> * penetration_locators =
+          NULL;
+
+      bool displaced = false;
+      _augLM_repeat_step = false;
+      if (getDisplacedProblem() == NULL)
+      {
+        GeometricSearchData & geom_search_data = geomSearchData();
+        penetration_locators = &geom_search_data._penetration_locators;
+      }
+      else
+      {
+        GeometricSearchData & displaced_geom_search_data = getDisplacedProblem()->geomSearchData();
+        penetration_locators = &displaced_geom_search_data._penetration_locators;
+        displaced = true;
+      }
+
+      for (const auto & it : *penetration_locators)
+      {
+        PenetrationLocator & pen_loc = *(it.second);
+
+        BoundaryID slave_boundary = pen_loc._slave_boundary;
+
+        if (constraints.hasActiveNodeFaceConstraints(slave_boundary, displaced))
+        {
+          const auto & ncs = constraints.getActiveNodeFaceConstraints(slave_boundary, displaced);
+
+          for (const auto & nc : ncs)
+          {
+            if (std::dynamic_pointer_cast<MechanicalContactConstraint>(nc) == NULL)
+              mooseError("AugmentedLagrangianContactProblem: dynamic cast of "
+                         "MechanicalContactConstraint object failed.");
+
+            if (!(std::dynamic_pointer_cast<MechanicalContactConstraint>(nc))
+                     ->AugmentedLagrangianContactConverged())
+            {
+              (std::dynamic_pointer_cast<MechanicalContactConstraint>(nc))
+                  ->updateAugmentedLagrangianMultiplier(false);
+              _augLM_repeat_step = true;
+              break;
+            }
+          }
+        }
+      }
+
+      // if (_augLM_repeat_step || _num_lagmul_iterations == 0)
+      if (_augLM_repeat_step)
+      { // force it to keep iterating
+        reason = MOOSE_NONLINEAR_ITERATING;
+        _console << "Augmented Lagrangian Multiplier needs updating\n " << std::endl;
```

Comment:
Get rid of the extra space after the newline.  Also, maybe get rid of the second newline.